tests/test__collection_api.py

```diff
@@ -5153,3 +5153,93 @@ def test__aggregate_date_to_string(self):
             self.db.collection.aggregate(
                 [{'$project': {'a': {'$dateToString': '10'}}}]
             )
+
+    def test__aggregate_array_to_object(self):
+        collection = self.db.collection
+        collection.insert_many([{
+            'items': [['a', 1], ['b', 2], ['c', 3], ['a', 4]],
+        }, {
+            'items': [{'k': 'a', 'v': 1}, {'k': 'b', 'v': 2},
+                      {'k': 'c', 'v': 3}, {'k': 'a', 'v': 4}],
+        }, {
+            'items': [],
+        }, {
+            'items': None,
+        }])
+
+        actual = collection.aggregate([
+            {'$project': {
+                'items': {'$arrayToObject': '$items'},
+                'not_exists': {'$arrayToObject': '$nothing'},
+                '_id': 0
+            }}
+        ])
+
+        expect = [{
+            'items': {'a': 4, 'b': 2, 'c': 3},
+            'not_exists': None
+        }, {
+            'items': {'a': 4, 'b': 2, 'c': 3},
+            'not_exists': None
+        }, {
+            'items': {},
+            'not_exists': None
+        }, {
+            'items': None,
+            'not_exists': None
+        }]
+        self.assertEqual(expect, list(actual))
+
+        with self.assertRaises(TypeError):
```

Comment:
Here you could use a table test: create a list of `items` that should trigger an error and then iterate over them and do an aggregation and an asserRaises for each of them.